src/api/dataSet.js

```diff
@@ -4,36 +4,35 @@ export async function getDataSet(dataSetId) {
   return await Ajax.get(`/dataset/${dataSetId}/`);
 }
 
-export async function getSamplesAndExperiments(dataSet) {
-  const experiments = {},
-    samples = {};
-
-  await Promise.all(
-    Object.keys(dataSet).map(async accessionCode => {
-      const experiment = await Ajax.get('/experiments/', {
-        accession_code: accessionCode
-      });
-
-      experiments[accessionCode] = experiment.results[0];
-
-      /*
-      Use sample accessions from dataSet instead of experiment
-      because some experiments may only have a subset of samples added
-      */
-      const sampleList = dataSet[accessionCode];
-      const response = await Ajax.get('/samples/', {
-        limit: 1000000000000000,
-        accession_codes: sampleList.join(',')
-      });
-      const sampleInfo = response.results;
-
-      samples[accessionCode] = sampleInfo;
-    })
-  );
-
-  return { experiments, samples };
+export async function getDataSetDetails(dataSetId) {
+  return await Ajax.get(`/dataset/${dataSetId}/`, {details: true});
 }
 
 export async function updateDataSet(dataSetId, dataSet) {
   return await Ajax.put(`/dataset/${dataSetId}/`, { data: dataSet });
 }
+
+
+
+// Takes in arrays of samples and experiments as formatted by the serializer
+// and turns them into objects where the keys are experiment accession codes
+
+export function formatExperiments(experiments) {
+  return experiments.reduce((accum, experiment) => {
+    accum[experiment.accession_code] = experiment;
+    return accum;
+  }, {});
+}
+
+export function formatSamples(data, samples) {
+  let result = {};
+
+  for (let accession_code in data) {
```

Comment:
I don't feel very comfortable about the `for/in` loop here, because it will iterate inherited properties too:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in
Although it is not wrong to use `for/in` here, I would prefer to iterate over `Object.keys(data)`, which will include direct properties only. What do you think, @arielsvn?

Reply:
You're right @dongbohu, thanks for reviewing! I'll update this later and let you know